src/components/network/selection-creation-panel.tsx

```diff
@@ -226,11 +277,10 @@ const FilterCreationPanel: React.FC<FilterCreationPanelProps> = ({
                         onClick={() => {
                             formMethods.trigger().then((isValid) => {
                                 if (isValid && defaultFolder) {
-                                    onSaveFilter(
-                                        formMethods.getValues() as IFilterCreation,
-                                        defaultFolder
-                                    );
-                                    generateFilterName();
+                                    const data =
+                                        formMethods.getValues() as IFilterSelection;
+                                    onSaveSelection(data, defaultFolder);
+                                    generateSelectionName(data[SELECTION_TYPE]);
                                 }
```

Comment:
react hook form already has a method doing all of this for you (checking validation error, getting all form values for you, ...)
Take a look at other forms for a working example

Reply:
this is an existing code, I just renamed it, Should I refactor code here ?

Reply:
it's due to the UniqueNameInput that not allow us to put default value